tests/D2L.CodeStyle.Analyzers.Test/Specs/ExternalDataSetVisibilityProviderAnalyzer.cs

```diff
@@ -0,0 +1,26 @@
+﻿// analyzer: D2L.CodeStyle.Analyzers.ApiUsage.DataHub.ExternalDataSetVisibilityProviderAnalyzer
+
+using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
+
+namespace D2L.AW.DataExport.BrightspaceDataSets.Domain {
+	internal interface IEventDrivenDataSetPlugin { }
+	internal interface IExternalDataSetVisibilityProvider<T> where T : IEventDrivenDataSetPlugin { }
+}
+
+namespace D2L.CodeStyle.Analyzers.Tests.Specs {

```

Comment:
It would be good for this to include classes which don't ` : IEventDrivenDataSetPlugin`, to show we won't raise diagnostics regardless of the `IExternalDataSetVisibilityProvider` injected.

Would be also be good to include a `IEventDrivenDataSetPlugin` class which _doesn't_ inject a `IExternalDataSetVisibilityProvider`